### ISSUE TITLE
Failing to parse something: "No possible parsings"

### ISSUE THREAD

**Issue by alecmocatta** [Hardmath123/nearley#59]
The following grammar fails to parse `$a;` with `No possible parsings (@2: ';')`.

A change - which shouldn't make any difference - lets it parse correctly. Changing the definition of `statement` to reference `product` rather than `sum` works – even through `sum -> product`.

```
program -> _ block {% function(d) { return d[1]; } %}
block -> (statement _):* {% function(d) { return ["block", d[0].map(function(s){return s[0];})]; } %}
#statement -> expression _ ";" {% id %}
statement -> sum _ ";" {% id %}
#statement -> product _ ";" {% id %}
expression -> expression _ ("="|"=="|"!="|">"|"<"|"<="|">=") _ sum {% function(d) { return ["operation", d[0], d[2], d[4]]; } %} | sum
#sum -> sum ("*"|"/") product | product
sum -> product
product -> product ("*"|"/") exp | exp
exp -> unaryoperation "^" exp | unaryoperation # this is right associative!
unaryoperation -> unaryoperation _ ("++"|"--") {% function(d) { return ["unaryoperation",d[0],d[2]]; } %} | mapoperation
mapoperation -> mapoperation _ "[" _ expression _ "]" {% function(d) { return ["map",d[0],d[4]]; } %} | element
element -> variable {% id %} | number {% id %} | "(" _ expression _ ")" {% function(d) { return d[2]; } %} | "{" _ block "}" {% function(d) { return d[2]; } %} | "if" _ expression _ expression {% function(d) { return ["if",d[2],d[4]]; } %} | "while" _ expression _ expression {% function(d) { return ["for",d[2],d[4]]; } %}
variable -> "$":? [a-z]:+ {% function(d) { return ["variable", d[0], d[1].join("")]; } %}
number -> [0-9]:+ {% function(d) { return ["number", d[0].join("")]; } %}
_ -> ___:* {% id %}
__ -> ___:+ {% id %}
___ -> [\t \n] {% empty %} | mlcomment | slcomment
mlcomment -> "/*" mlcommentchars:+ .:? "*/" {% function(d) { return [];/*["comment", d[1].join("")+d[2]];*/ } %}
slcomment -> "//" [^\n]:* "\n" {% function(d) { return [];/*["comment", d[1].join("")];*/ } %}
mlcommentchars -> "*" [^/] {% function(d) { return d[0] + d[1]; } %} | [^*] . {% function(d) { return d[0] + d[1]; } %}
```

**Comment by Hardmath123** [comment]
Ah, I see what's causing this—some weird corner-case subtlety with the memoization going on in nullables. Grr. I'll have it fixed ASAP.

**Comment by Hardmath123** [comment]
Update: should be fixed in the latest nearley version (v1.3.1).
